### DISCUSSION TITLE
Shouldn't org.infinispan.commons.util.ClasspathURLStreamHandler throw an exception instead of returning null, if a resource is not found?

### DISCUSSION THREAD

**Discussion by gurka** [infinispan/infinispan#10763] [Q&A] [answered]
Hi,

We are having some issue upgrading infinispan from version 9 to 13 in our application. I've posted a more [general question](https://stackoverflow.com/questions/75877254/conflict-when-both-infinispan-and-qpid-broker-registers-urlstreamhandlers) on stackoverflow which unfortunately hasn't received any answers yet, 

[This PR](https://github.com/infinispan/infinispan/pull/9074) adds a ClasspathURLStreamHandler, which can automatically register itself globally to the JVM. This is causing issue for us in our application, as it happens to register itself before another library (`qpid-broker`) tries to register its handler. The actual issue seems to be that these two handlers have different behaviour when the requested resource cannot be found. infinispan's handler returns null while qpid-broker's handler throws an exception (`FileNotFoundException` or `MalformedURLException`).

- Shouldn't infinispan's handler also throw an appropriate exception on error, instead of simply returning null?
- Is it possible to disable automatic registration of this handler in infinispan?

**Comment by tristantarrant** [comment, marked as answer]
Hi @gurka , I've created https://issues.redhat.com/browse/ISPN-14732

**Reply by gurka** [reply]
Thank you!
